actionpack/lib/action_dispatch/middleware/params_parser.rb

```diff
@@ -1,15 +1,16 @@
 require 'active_support/core_ext/hash/conversions'
 require 'action_dispatch/http/request'
 require 'active_support/core_ext/hash/indifferent_access'
+require "active_support/core_ext/module/delegation"
 
 module ActionDispatch
   class ParamsParser
     class ParseError < StandardError
-      attr_reader :original_exception
+      delegate :to_s, to: :cause
 
-      def initialize(message, original_exception)
-        super(message)
-        @original_exception = original_exception
```

Comment:
`#cause` will return `nil` unless the exception is actually raised. Technically we can use `$!`  instead which holds the same exception as Ruby internally captures `$!` and set it to `#cause`. But still extra interface to do one thing.